### PR TITLE
Run cargo-fmt on `datafusion-functions/core`

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/9281

## Rationale for this change

`cargo fmt` is not run on much of the code in datafusion-functions due to https://github.com/rust-lang/rustfmt/issues/3253, As @jonahgao found

 This makes PRs larger than necessary and misses all the benefits of cargo-fmt

## What changes are included in this PR?

1. Change how the `datafusion-functions/core` module is declared using a direct conditional include so it it is visible to rust-fmt
2. Run `cargo fmt` and check in the resulting formatting changes in core

## Are these changes tested?
Yes, by CI

## Are there any user-facing changes?
No, this is an internal development process change

### CODE REVIEW

File: datafusion/functions/src/core/nullif.rs [hunk 1]
```diff
@@ -18,15 +18,15 @@
...
+use datafusion_common::{internal_err, DataFusionError, Result};
 use datafusion_expr::ColumnarValue;
 
-use datafusion_expr::{ScalarUDFImpl, Signature, Volatility};
-use std::any::Any;
 use arrow::array::Array;
 use arrow::compute::kernels::cmp::eq;
 use arrow::compute::kernels::nullif::nullif;
 use datafusion_common::ScalarValue;
+use datafusion_expr::{ScalarUDFImpl, Signature, Volatility};
```
Comment (alamb) [inline datafusion/functions/src/core/nullif.rs:28]: The changes in this file are entirely due to `cargo-fmt` being run now

File: datafusion/functions/src/lib.rs [hunk 2]
```diff
@@ -84,7 +84,11 @@ use log::debug;
 #[macro_use]
 pub mod macros;
 
-make_package!(core, "core_expressions", "Core datafusion expressions");
+/// Core datafusion expressions
+/// Enabled via feature flag `core_expressions`
```
Comment (alamb) [inline datafusion/functions/src/lib.rs:88]: This is the actual change. If this PR is merged, I will make follow on PR(s) to apply the same treatment to the other modules in `datafusion-functions`. 

I did only one initially to keep the size of the PR down

File: datafusion/functions/src/macros.rs [hunk 3]
```diff
@@ -122,6 +122,30 @@ macro_rules! make_package {
     };
 }
 
+/// Macro creates a sub module if the feature is not enabled
+///
+/// The rationale for providing stub functions is to help users to configure datafusion
+/// properly (so they get an error telling them why a function is not available)
+/// instead of getting a cryptic "no function found" message at runtime.
+
+macro_rules! make_stub_package {
```
Comment (alamb) [inline datafusion/functions/src/macros.rs:131]: This is part of the `make_package` macro which I think keeps most of the "don't repeat yourself" goal but still allows `cargo fmt` to run.

Once I port over the other modules to use this pattern,  I will remove the `make_package` macro

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb and @jonahgao 

so package level macros prevented `cargo fmt` to run?

**Review by Jefffrey — APPROVED** [review]
Nice :+1:

**Comment by alamb** [comment]
Here is the follow on PR to run rustfmt on all the modules: https://github.com/apache/arrow-datafusion/pull/9390

**Pull request merged**